components/SubmissionComments.tsx

```diff
@@ -1,11 +1,44 @@
-import React from 'react'
+import React, { useContext } from 'react'
 import { Comment, Submission } from '../graphql'
 import Markdown from 'markdown-to-jsx'
 import ReviewerProfile from './ReviewerProfile'
+import { Button } from './theme/Button'
+import { TrashIcon } from '@primer/octicons-react'
+import styles from '../scss/submissionComments.module.scss'
+import { updateCache } from '../helpers/updateCache'
+import { useDeleteCommentMutation } from '../graphql'
+import { GlobalContext } from '../helpers/globalContext'
+
 export const SubmissionComments: React.FC<{
   comments: Comment[]
   submission: Submission
-}> = ({ comments }) => {
+}> = ({ comments, submission }) => {
+  const [deleteComment] = useDeleteCommentMutation()
+  const context = useContext(GlobalContext)
+  const id = context.session?.user?.id
+
+  const handleDeleteComment = (c: Comment) => {
+    const update = updateCache(
+      submission.id,
+      c.id,
+      undefined,
```

Comment:
So many undefined arguments passed through. This is pretty ugly

Reply:
https://github.com/garageScript/c0d3-app/pull/1474